src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if ((object.getClass().equals(Rectangle.class))) {
+            Rectangle rectangle = (Rectangle) object;
+            return (rectangle.length == null && length == null
+                    || length.compareTo(rectangle.length) == 0)
```

Comment:
1)You have a potential NPE risk here when length == null, but rectangle.length is not;
2) use equals() instead of compareTo